xmppserver/src/main/java/org/jivesoftware/openfire/pep/IQPEPHandler.java

```diff
@@ -314,40 +313,17 @@ public IQ handleIQ(IQ packet) throws UnauthorizedException {
             packet.setTo(jidFrom);
 
             if (packet.getType() == IQ.Type.set) {
```

Comment:
Probably a style preference more than anything else;
a) I'd like to see a `switch(packet.getType())` as is should make it clearer as to which types are being handled. I /think/ the `packet.getTo() == null` branch only handles a IQ Set or Get - what's the switch default condition? I think it's the final `return null` of the current method, but would be good to make that really explicit. 
b) Similarly, is it possible to break up the different branches into methods on their own - so we have small methods `handleIQSetToNull(...)`, `handleIQGetToNull(...)`, `handleIQRequest(...)` etc.

Reply:
I initially thought this to be a bit of overkill, but I actually found a small issue while doing this: https://issues.igniterealtime.org/browse/OF-1992